src/assign.c

```diff
@@ -930,48 +929,55 @@ const char *memrecycle(SEXP target, SEXP where, int start, int len, SEXP source,
 
 #define COERCE_ERROR(targetType) error("type '%s' cannot be coerced to '%s'", type2char(TYPEOF(source)), targetType); // 'targetType' for integer64 vs double
 
-  const int off = (length(where)?0:start); // off = target offset; e.g. called from rbindlist with where=R_NilValue and start!=0
-  const bool mem = length(where)==0 && slen==len;  // only used if types match too
+  const int off = length(where) ? 0 : start;  // off = target offset; e.g. called from rbindlist with where=R_NilValue and start!=0
+  const bool mc = length(where)==0 && slen==len;  // mc=memcpy; only used if types match too
+  const int *wd = length(where) ? INTEGER(where)+start : NULL;
```

Comment:
wd seems to be used only in string branch, why not to move that line there?

Reply:
wd is used in BODY macro